pkg/collector/clock.go

```diff
@@ -82,6 +82,10 @@ func (t *fakeTimer) Reset(d time.Duration) bool {
 		fired = false
 		t.targetTime = clock.now.Add(d)
 	}
+	if fired {
+		t.targetTime = clock.now.Add(d)
```

Comment:
Looks like we need to update the targetTime whether we found the timer or not? Do you think it's better to move it out of the bracket?